pkg/hvs/controllers/flavortemplate_controller.go

```diff
@@ -0,0 +1,327 @@
+	// Check whether the template is adhering to the schema
+	schemaLoader := gojsonschema.NewSchemaLoader()
+
+	definitionsSchemaJSON, err := readJSON(ftc.CommonDefinitionsSchema)
+	if err != nil {
+		return "Unable to read the schema", errors.Wrap(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Unable to read the file"+consts.CommonDefinitionsSchema)
+	}
+
+	definitionsSchema := gojsonschema.NewStringLoader(definitionsSchemaJSON)
+	templateSchemaJSON, err := readJSON(ftc.FlavorTemplateSchema)
+	if err != nil {
+		return "Unable to read the schema", errors.Wrap(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Unable to read the file"+consts.FlavorTemplateSchema)
+	}
+	flvrTemplateSchema := gojsonschema.NewStringLoader(templateSchemaJSON)
+	schemaLoader.AddSchemas(definitionsSchema)
+
+	schema, err := schemaLoader.Compile(flvrTemplateSchema)
+	if err != nil {
+		return "Unable to Validate the template", errors.Wrap(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Unable to compile the schemas")
+	}
+
+	documentLoader := gojsonschema.NewStringLoader(template)
+
+	result, err := schema.Validate(documentLoader)
+	if err != nil {
+		return "Unable to Validate the template", errors.Wrap(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Unable to validate the template")
+	}
+
+	var errorMsg string
+	if !result.Valid() {
+		for _, desc := range result.Errors() {
+			errorMsg = errorMsg + fmt.Sprintf("- %s\n", desc)
+		}
+		return errorMsg, errors.New("controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() The provided template is not valid" + errorMsg)
+	}
+
+	defaultLog.Info("controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() The provided template is valid")
+
+	//Validation the syntax of the conditions
+	tempDoc, err := jsonquery.Parse(strings.NewReader("{}"))
+	if err != nil {
+		return "", errors.Wrap(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Unable to parse json query")
+	}
+
+	for _, condition := range FlvrTemp.Condition {
+		_, err := jsonquery.Query(tempDoc, condition)
+		if err != nil {
+			return "Invalid syntax in condition statement", errors.Wrapf(err, "controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Invalid syntax in condition : %s", condition)
+		}
+	}
+
+	//Check whether each pcr index is associated with not more than one bank.
+	pcrMap := make(map[*hvs.FlavorPart][]hvs.PCR)
+	flavors := []*hvs.FlavorPart{FlvrTemp.FlavorParts.Platform, FlvrTemp.FlavorParts.OS, FlvrTemp.FlavorParts.HostUnique, FlvrTemp.FlavorParts.Software}
+	for _, flavor := range flavors {
+		if flavor != nil {
+			if _, ok := pcrMap[flavor]; !ok {
+				var pcrs []hvs.PCR
+				for _, pcrRule := range flavor.PcrRules {
+					pcrs = append(pcrs, pcrRule.Pcr)
+				}
+				pcrMap[flavor] = pcrs
```

Comment:
Can we validate the condition at #306 here only?

Reply:
Getting all the pcrs given for the particular flavor, then evaluating whether any duplicates are present.  Do we need to change that ?

Reply:
looks good in that case